Danki2/Assets/Scripts/Stats/StatsManager.cs

```diff
@@ -1,27 +1,43 @@
 ﻿using System;
+using System.Collections.Generic;
 
 public class StatsManager
 {
     private readonly Stats _baseStats;
-    private readonly Stats _frameStats;
+    private readonly Dictionary<Stat, float> _statsCache;
+    private List<Func<Stat, float, float>> _pipes;
 
     public StatsManager(Stats baseStats)
     {
         _baseStats = baseStats;
-        _frameStats = new Stats(_baseStats);
+        _statsCache = new Dictionary<Stat, float>();
+        _pipes = new List<Func<Stat, float, float>>();
     }
 
-    public int this[Stat stat]
+    public float this[Stat stat]
     {
-        get { return _frameStats[stat]; }
-        set { _frameStats[stat] = value; }
+        get {
+            if (_statsCache.TryGetValue(stat, out float value))
+            {
+                return value;
+            }
+            else
+            {
+                var pipelineValue = (float)_baseStats[stat];
+                _pipes.ForEach(p => pipelineValue = p(stat, pipelineValue));
+                _statsCache.Add(stat, pipelineValue);
+                return pipelineValue;
+            }
+        }
     }
 
-    public void Rebase()
+    public void RegisterPipe(Func<Stat, float, float> pipe)
     {
-        foreach (Stat stat in Enum.GetValues(typeof(Stat)))
-        {
-            _frameStats[stat] = _baseStats[stat];
-        }
+        _pipes.Add(pipe);
+    }
+
+    public void ResetCache()
```

Comment:
Another small point, but would `ClearCache()` be a better name? To me it feels like you clear a cache rather than reset it, but happy to leave as is if you prefer